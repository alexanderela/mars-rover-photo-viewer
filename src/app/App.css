#root {
  max-width: 1280px;
  margin: 0;
  text-align: center;
  width: 100%;
}

body {
  margin: 0;
}
