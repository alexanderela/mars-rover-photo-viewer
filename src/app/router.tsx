import {
  createBrowserRouter,
  redirect,
} from "react-router-dom";
import { EnlargedPhotoViewer } from "../features/EnlargedPhotoViewer";

export const routesConfig = [
  {
    path: "/",
    lazy: async () => {
      const { default: App } = await import("./App");

      return {
        element: (
          <App />
        ),
      }
    },
    children: [
      {
        index: true,
        loader: async () => redirect("rovers/curiosity?page=1")
      },
      {
        path: "rovers/:rover",
        lazy: async () => {
          const { PhotoViewer } = await import("../features/photos/PhotoViewer");
    
          return {
            element: (
              <PhotoViewer />
            )
          }
        },
        children: [
          {
            path: "id/:id",
            element: <EnlargedPhotoViewer />
          }
        ]
      },
    ]
  },
];

export const router = createBrowserRouter(routesConfig);