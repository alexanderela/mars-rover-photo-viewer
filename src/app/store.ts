import { configureStore } from "@reduxjs/toolkit";
import roverPhotosReducer from "../reducers/roverPhotosSlice";
import pageSliceReducer from "../reducers/pageSlice";
import selectedRoverSliceReducer from "../reducers/selectedRoverSlice";

const reducer = {
  roverPhotos: roverPhotosReducer,
  page: pageSliceReducer,
  selectedRover: selectedRoverSliceReducer,
}

export const store = configureStore({ 
  reducer,
});

export type AppStore = typeof store;
export type RootState = ReturnType<AppStore['getState']>;
export type AppDispatch = AppStore['dispatch'];