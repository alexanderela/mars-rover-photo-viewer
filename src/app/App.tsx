import Grid from "@mui/material/Grid2";
import { Header } from "../layout/Header";
import { SideNav } from "../layout/SideNav";
import { Content } from "../layout/Content";
import "./App.css";

const App = () => {
  return (
    <>
      <Grid
        container
        sx={{
          backgroundColor: (theme) => theme.palette.primary.main,
          height: "100%",
          width: "100vw",
          padding: 0,
        }}
      >
        <Header />
        <SideNav />
        <Content />
      </Grid>
    </>
  );
};

export default App;
