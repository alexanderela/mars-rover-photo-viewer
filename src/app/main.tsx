import { createRoot } from "react-dom/client";
import { Provider } from "react-redux";
import { ThemeProvider } from "@mui/material/styles";
import { theme } from "../styles/theme.ts";
import { setupStore } from "./store.ts";
import { RouterProvider } from "react-router-dom";
import { router } from "./router.tsx";
import "../index.css";
import "../reset.css";

createRoot(document.getElementById("root")!).render(
  <Provider store={setupStore()}>
    <ThemeProvider theme={theme}>
      <RouterProvider router={router} />
    </ThemeProvider>
  </Provider>,
);
