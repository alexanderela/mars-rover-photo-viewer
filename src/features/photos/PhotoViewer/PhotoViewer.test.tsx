import { describe, expect, it } from "vitest";
import { render, waitFor } from "../../../tests/test-utils";
import { createMemoryRouter,  RouterProvider } from "react-router-dom";
import { routesConfig } from "../../../app/router";

describe("<PhotoViewer", () => {
  it("renders PhotoViewer component", async () => {
    const router = createMemoryRouter(routesConfig, {
      initialEntries: ["/rovers/curiosity?page=1"],
    });

    const result = render(<RouterProvider router={router} />);

    await waitFor(() => {
      const element = result.getByTestId(/PhotoViewer/i);
      expect(element).toBeInTheDocument();
      expect(element).toMatchSnapshot();
    })
  });
});