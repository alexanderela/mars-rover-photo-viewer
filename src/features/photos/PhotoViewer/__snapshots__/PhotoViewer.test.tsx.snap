// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<PhotoViewer > renders PhotoViewer component 1`] = `
<div
  class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-irsnvt-MuiGrid2-root"
  data-testid="PhotoViewer"
/>
`;
